### ISSUE TITLE
Modify neighbor via API without requiring config reread

### ISSUE THREAD

**Issue by thepacketgeek** [Exa-Networks/exabgp#294]
I can see a scenario happening where a BGP neighbor needs to be add/removed from ExaBGP's peering without affecting the other neighbors. Currently, to add/remove/shutdown a peer the ExaBGP process must be restart to reread the config file.

An API command similar to the existing commands can be added, something like:
    ```neighbor add 10.10.0.1 peer-as 65000 [local-as 65001 | local-address 10.0.0.1]```
    ```neighbor shutdown 10.10.0.1```
    ```neighbor resume 10.10.0.1```
    ```neighbor remove 10.10.0.1```

**Comment by thomas-mangin** [comment]
Yes , this feature has been requested multiple time and is on the top of the todo list.

**Comment by thepacketgeek** [comment]
I just tested and realized the 'reload' API command does not tear down existing BGP sessions. This works for me as I can just issue a reload command when I need to add/remove a peer.

**Comment by thomas-mangin** [comment]
😀

**Comment by doddt** [comment]
I am also interested in this feature, and see this discussion is a year old. I'm running 3.4.16 but don't see any way to add/delete/modify peers without a config reload. Is this enhancement coming soon?

**Comment by thomas-mangin** [comment]
It is coming .. saying soon would surely be overselling it .. I am short of open source coding time ..

**Comment by mikrologic** [comment]
Great work on exabgp Thomas!    Just curious as to the status of: Modify neighbor via API without requiring config reread #294       Thanks!

**Comment by thomas-mangin** [comment]
@mikrologic nothing changed, still one of the feature I want to see soon. It would use a template as reference, not allow free format but I tend to work on ExaBGP when I have some free time and it is short in supply atm .. so no idea.

**Comment by mikrologic** [comment]
Thanks for the info I appreciate it.  And I understand the lack of time 
issue.     One thing I would love to get your opinion on is the use of 
Exa to distribute IP blocklists.  I managed to do it with 
https://github.com/infowolfe/exabgp-edgerouter    However it takes 10 
minutes to load a medium sized list.  Is there any way to have Exa 
utilize MySQL to load neighbors and routes?     Thanks again.


Regards,
                   Josh

**Comment by thomas-mangin** [comment]
Sure - you can use whatever backend and then optimise the output ... but this is a different question - perhaps best on the ML.

**Comment by mikrologic** [comment]
Do you or someone you know do any custom work?   I need to figure out 
how to quickly load large blocklists and serve to neighbors.  The 
current methods I have tried take 1.5 Hrs to load
40k IPs.   If so I would be interested in rates.


Thanks

**Comment by iddq** [comment]
Is there any shutdown/no shutdown option?
